### PR TITLE
Expand documentation

### PR DESCRIPTION
**Pull request by rpgoldman**
Add a note to explain that backend creation must be done in a model context.
Also, amplify the discussion of `Bound`.  TBH, I'm not sure I fully understand what sort of thing a `Bound` is.  It *contains* a distribution, but it is actually used to create another random variable.  Perhaps it should be referred to as a `Callable`?
I can *definitely* refine this PR before it's merged.

### CODE REVIEW

File: pymc3/backends/__init__.py [hunk 1]
```diff
@@ -20,8 +20,12 @@
...
 
     >>> import pymc3 as pm
-    >>> db = pm.backends.Text('test')
-    >>> trace = pm.sample(..., trace=db)
+    >>> with pm.Model():
+    >>>      db = pm.backends.Text('test')
+    >>>      trace = pm.sample(..., trace=db)
+
+Note that as in the example above, one must have an active model context,
+or pass a `model` parameter in order to create a backend.
```
Comment (fonnesbeck) [inline pymc3/backends/__init__.py:28]: More fundamentally than that, you can only create a new backend in a call to `sample` (which is always called in a Model context)

Comment (rpgoldman) [reply pymc3/backends/__init__.py:28]: I don't actually understand your claim here.  The assignment to `db` in the code snippet above works, and is not in a call to `sample`.  So it seems to me that I am creating a backend outside the call to `sample`, and just *populating* it in the call to `sample`.  Is that not correct?  If not, what happens in line 24 above?  How should we describe that, if it's not creating a new backend?  Thanks.

Comment (fonnesbeck) [reply pymc3/backends/__init__.py:28]: You are correct. I completely misread what you wrote. Apologies.

**Comment by fonnesbeck** [comment]
LGTM

**Comment by junpenglao** [comment]
Thanks @rpgoldman !

**Comment by rpgoldman** [comment]
@ColCarroll You said I should ping you about setting up a documentation build test ground.... Consider yourself pinged! If it's convenient, I think you can reach me by email through my github account.

**Comment by ColCarroll** [comment]
wait, but not on this pr, right? is there a version of the docs you'd like to see?

**Comment by ColCarroll** [comment]
current master is built -- your `Multitrace` doc PR is available here: https://pymc3.colindcarroll.com/api/inference.html?highlight=multitrace#multitrace

**Comment by rpgoldman** [comment]
@ColCarroll Thanks! I think I misunderstood your response on the discourse site.

**Pull request merged**